sw/device/tock/dif/build.rs

```diff
@@ -0,0 +1,62 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+extern crate bindgen;
+
+use bindgen::builder;
+use std::env;
+use std::fs::{self};
+use std::path::Path;
+
+fn main() {
+    let source_root = env::var("MESON_SOURCE_ROOT").unwrap();
+    let build_root = env::var("MESON_BUILD_ROOT").unwrap();
+    let out_dir = env::var("OUT_DIR").unwrap();
+
+    generate_bindings(&source_root, &build_root, &out_dir);
+
+    let dif_lib_path = Path::new(&build_root).join("sw/device/lib/dif");
+    let dif_libs = &["uart_ot"];
+    libraries_link(dif_lib_path.to_str().unwrap(), dif_libs);
+
+    let mmio_lib_path = Path::new(&build_root).join("sw/device/lib/base");
+    let base_libs = &["mmio_ot"];
+    libraries_link(mmio_lib_path.to_str().unwrap(), base_libs);
+}
+
+fn libraries_link(path: &str, libs: &[&str]) -> () {
+    for lib in libs {
+        println!("cargo:rustc-link-search={}", path);
+        println!("cargo:rustc-link-lib={}", lib);
+    }
+}
+
+fn generate_bindings(source_root: &String, build_root: &String, out_dir: &String) {
+    let mut binding_builder = builder()
+        .raw_line("use riscv32_c_types;")
+        .ctypes_prefix("riscv32_c_types")
+        .clang_arg("--target=riscv32")
+        .clang_args(&["-I", &source_root])
+        .clang_args(&["-I", &build_root]);
+
+    let dif_path = Path::new(&source_root).join("sw/device/lib/dif");
+    let dif_build_meson = dif_path.join("meson.build");
+    println!("cargo:rerun-if-changed={}", dif_build_meson.to_str().unwrap());
+
+    let files = fs::read_dir(dif_path).unwrap();
+    for file in files {
+        let path = file.unwrap().path();
+        if let Some(e) = path.extension() {
+            let path = path.to_str().unwrap();
```

Comment:
Maybe don't name this the same as `path` above?

Reply:
Done.